logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -15,6 +15,21 @@ def close
   end
 end
 
+class DummyInputGenerator < LogStash::Inputs::Base
```

Comment:
are we already ok on going away from plugins to be used in specs? I don't think we agreed, even if I love the idea. not sure what to think.